lib/diaspora/federation/receive.rb

```diff
@@ -0,0 +1,322 @@
+module Diaspora
+  module Federation
+    module Receive
+      extend Diaspora::Logging
+
+      def self.account_deletion(entity)
+        AccountDeletion.create!(person: author_of(entity), diaspora_handle: entity.author)
+      end
+
+      def self.comment(entity)
+        receive_relayable(Comment, entity) do
+          Comment.new(
+            author:      author_of(entity),
+            guid:        entity.guid,
+            created_at:  entity.created_at,
+            text:        entity.text,
+            commentable: Post.find_by(guid: entity.parent_guid)
+          )
+        end
+      end
+
+      def self.contact(entity)
+        recipient = Person.find_by(diaspora_handle: entity.recipient).owner
+        if entity.sharing.to_s == "true"
```

Comment:
Note to myself: No, @SuperTux88 is not crazy. The federation gem is only capable of passing strings at the moment and the conversion to a Boolean happens by AR. This is the only place where a check of that kind is needed, so the adjustment to the federation gem has a low priority, but will be done eventually.